account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -0,0 +1,368 @@
+    username = fields.Char()
+    password = fields.Char()
+    key = fields.Binary()
+    certificate_public_key = fields.Text()
+    certificate_private_key = fields.Text()
+    certificate_chain = fields.Text()
+
+    _sql_constraints = [
+        (
+            'journal_id_uniq',
+            'UNIQUE(journal_id)',
+            'Only one online banking statement provider per journal!'
+        ),
+        (
+            'valid_interval_number',
+            'CHECK(interval_number > 0)',
+            'Scheduled update interval must be greater than zero!'
+        )
+    ]
+
+    @api.model
+    def _get_available_services(self):
+        """Hook for extension"""
+        return []
+
+    @api.model
+    def _selection_service(self):
+        return self._get_available_services() + [('dummy', 'Dummy')]
+
+    @api.model
+    def values_service(self):
+        return self._get_available_services()
+
+    @api.multi
+    @api.depends('service')
+    def _compute_name(self):
+        for provider in self:
+            provider.name = list(filter(
+                lambda x: x[0] == provider.service,
+                self._selection_service()
+            ))[0][1]
+
+    @api.multi
+    @api.depends('active', 'interval_type', 'interval_number')
+    def _compute_update_schedule(self):
+        for provider in self:
+            if not provider.active:
+                provider.update_schedule = _('Inactive')
+                continue
+
+            provider.update_schedule = _('%(number)s %(type)s') % {
+                'number': provider.interval_number,
+                'type': list(filter(
+                    lambda x: x[0] == provider.interval_type,
+                    self._fields['interval_type'].selection
+                ))[0][1],
+            }
+
+    @api.multi
+    def _pull(self, date_since, date_until):
+        AccountBankStatement = self.env['account.bank.statement']
+        is_scheduled = self.env.context.get('scheduled')
+        if is_scheduled:
+            AccountBankStatement = AccountBankStatement.with_context(
+                tracking_disable=True,
+            )
+        AccountBankStatementLine = self.env['account.bank.statement.line']
+        for provider in self:
+            statement_date_since = provider._get_statement_date_since(
+                date_since
+            )
+            while statement_date_since < date_until:
+                statement_date_until = (
+                    statement_date_since + provider._get_statement_date_step()
+                )
+                try:
+                    data = provider._obtain_statement_data(
+                        statement_date_since,
+                        statement_date_until
+                    )
+                except Exception as e:
+                    _logger.warning(
+                        'Online Bank Statement Provider failure: %s' % e,
+                        exc_info=True,
+                    )
+                    provider.message_post(
```

Comment:
When there's a failure, the error msg is logged in the provider chatter. But the importing functionality is accessible from the bank account or the bank journal. So in case of failure, the user is not aware of it (no error displayed). This is confusing and requires the user to switch to the provider form view or to manually verify if the statements are imported.

Reply:
Fixed and rebased via #212